File: c++/Car.cpp
#include <iostream>
#include "Car.h"

using namespace std;

Car::Car(int vehicleId, string vehicleNumber, int capacity, bool automatic)
    : Vehicle(vehicleId, vehicleNumber, capacity)
{
    this->automatic = automatic;
}

void Car::displayVehicleDetails()
{
    cout << "\n----------- Car Details -----------" << endl;
    cout << "Vehicle ID : " << vehicleId << endl;
    cout << "Vehicle Number : " << vehicleNumber << endl;
    cout << "Capacity : " << capacity << endl;
    cout << "Automatic : " << automatic << endl;
}

File: java/Car.java
// inheritance 
// car class inherits proprties like vehicleid vehicleNumber and capacity from vehicle class
public class Car extends Vehicle {
// van car has its own property as a automatic
    private boolean automatic;

    public Car(int vehicleId, String vehicleNumber, int capacity, boolean automatic) {

        super(vehicleId, vehicleNumber, capacity);
        this.automatic = automatic;
    }

    @Override
    public void displayVehicleDetails() {

        System.out.println("\n----------- Car Details -----------");
        System.out.println("Vehicle ID : " + vehicleId);
        System.out.println("Vehicle Number : " + vehicleNumber);
        System.out.println("Capacity : " + capacity);
        System.out.println("Automatic : " + automatic);
    }
}
